### PR TITLE
Fix Email Authentication Issue Caused by Trailing Spaces

### LINKED ISSUE #518: Bug: Email Authentication Failure Due to Trailing Space in Input
When a user enters their email and password in the login text fields, and unintentionally adds a space at the end of the email, the space is erroneously included in the email input. Consequently, this leads to an email mismatch during authentication, resulting in login failures.

**Expected Behavior**
The system should recognize and trim any trailing spaces in the email input to ensure accurate authentication.

https://github.com/zulip/zulip-flutter/assets/97253470/28cc536d-2da7-418a-8d3a-161c7c2fd9a6

### PR DESCRIPTION
**Pull request by Mehul-Kumar-27**
Fixes #518

### CODE REVIEW

**Comment by Mehul-Kumar-27** [comment]
@gnprice, sir, I have closed the older MR (#519) and created this one with the necessary changes.
Sorry for the delay :)

File: test/widgets/login_test.dart [hunk 1]
```diff
@@ -113,6 +113,28 @@ void main() {
...
+      );
+
+      await tester.tap(findSubmitButton);
+      checkFetchApiKey(username: eg.selfAccount.email, password: 'p455w0rd');
+      await tester.idle();
+
+      check(testBinding.globalStore.accounts).single.equals(eg.selfAccount
+          .copyWith(id: testBinding.globalStore.accounts.single.id));
+    });
     // TODO test validators on the TextFormField widgets
```
Comment (gnprice) [inline test/widgets/login_test.dart:138]: nit: same as https://github.com/zulip/zulip-flutter/pull/519#discussion_r1496417321

**Review by gnprice — COMMENTED** [review]
Thanks for the revision! One comment from the previous PR still applies — see below.

BTW rather than close the PR and open a new one, it's better to update the existing PR because that keeps the review thread in one place. See our guide:
https://zulip.readthedocs.io/en/latest/git/pull-requests.html#update-a-pull-request
https://zulip.readthedocs.io/en/latest/git/using.html#force-push-changes-to-github-after-you-ve-altered-your-history

File: test/widgets/login_test.dart (hunk 1 above)
Comment (Mehul-Kumar-27) [reply test/widgets/login_test.dart]: Done Sir.

**Comment by gnprice** [comment]
Cool. The code changes now all look good, but the branch structure needs to be brought back into our style as described at https://github.com/zulip/zulip-flutter/pull/519#pullrequestreview-1883589525.

Oh also: the commit message on the first/main commit:

d5a1b2bfe login: Added the functionality to trim the whilespaces in the email. Also added the test case for the same logic.

is close, but needs some adjustments to be fully within our style. I can make a final pass on it before merging, but please try another round of editing it yourself too. See this section in particular: https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html#commit-summary-part-2

**Comment by chrisbobbe** [comment]
Merged, with those fixups. Thanks Mehul and Greg!

**Pull request merged**